app/components/autocomplete/emoji_suggestion/emoji_suggestion.tsx

```diff
@@ -0,0 +1,256 @@
+        },
+        listView: {
+            paddingTop: 16,
+            backgroundColor: theme.centerChannelBg,
+            borderRadius: 4,
+        },
+        row: {
+            flexDirection: 'row',
+            alignItems: 'center',
+            overflow: 'hidden',
+            paddingBottom: 8,
+            paddingHorizontal: 16,
+            height: 40,
+        },
+        container: {
+            paddingBottom: 12,
+        },
+    };
+});
+
+const keyExtractor = (item: string) => item;
+
+type Props = {
+    cursorPosition: number;
+    customEmojis: CustomEmojiModel[];
+    maxListHeight: number;
+    updateValue: (v: string) => void;
+    onShowingChange: (c: boolean) => void;
+    rootId: string;
+    value: string;
+    nestedScrollEnabled: boolean;
+    skinTone: string;
+    hasFilesAttached: boolean;
+}
+const EmojiSuggestion = ({
+    cursorPosition,
+    customEmojis = [],
+    maxListHeight,
+    updateValue,
+    onShowingChange,
+    rootId,
+    value,
+    nestedScrollEnabled,
+    skinTone,
+    hasFilesAttached,
+}: Props) => {
+    const theme = useTheme();
+    const style = getStyleFromTheme(theme);
+    const serverUrl = useServerUrl();
+    const flatListStyle = useMemo(() =>
+        [style.listView, {maxHeight: maxListHeight}]
+    , [style, maxListHeight]);
+
+    const searchTimeout = useRef<NodeJS.Timeout>();
+
+    const emojis = useMemo(() => {
+        const emoticons = new Set<string>();
+        for (const [key, index] of EmojiIndicesByAlias.entries()) {
+            const skin = getSkin(Emojis[index]);
+            if (!skin || skin === skinTone) {
+                emoticons.add(key);
+            }
+        }
+
+        for (const custom of customEmojis) {
+            emoticons.add(custom.name);
+        }
+
+        return Array.from(emoticons);
+    }, [skinTone, customEmojis]);
+
+    const searchTerm = useMemo(() => {
+        const match = value.substring(0, cursorPosition).match(EMOJI_REGEX);
+        return match?.[3] || '';
+    }, [value, cursorPosition]);
+
+    const fuse = useMemo(() => {
+        return new Fuse(emojis, FUSE_OPTIONS);
+    }, [emojis]);
+
+    const data = useMemo(() => {
+        const searchTermLowerCase = searchTerm.toLowerCase();
+
+        if (searchTerm.length < MIN_SEARCH_LENGTH) {
+            return [];
+        }
+
+        const sorter = (a: string, b: string) => {
+            return compareEmojis(a, b, searchTermLowerCase);
+        };
+
+        const fuzz = fuse.search(searchTermLowerCase);
+
+        if (fuzz) {
+            const results = fuzz.reduce((values, r) => {
+                const score = r?.score === undefined ? 1 : r.score;
+                const v = r?.matches?.[0]?.value;
+                if (score < 0.2 && v) {
+                    values.push(v);
+                }
+
+                return values;
+            }, [] as string[]);
+
+            return results.sort(sorter);
+        }
+
+        return [];
```

Comment:
Duplicate of https://github.com/mattermost/mattermost-mobile/blob/55e55ca45878e5365ba0b03c603301e24e1bd66f/app/components/emoji_picker/filtered/index.tsx#L52